File: go/examples/fizzbuzz.go
package main

import (
  "fmt"
)

func main() {
  for i := 1; i <= 100; i++ {
    if dividesBy(i,15) {
      fmt.Println("FizzBuzz")
    } else if dividesBy(i,3) {
      fmt.Println("Fizz")
    } else if dividesBy(i,5) {
      fmt.Println("Buzz")
    } else {
      fmt.Printf("%d\n", i)
    }
  }
}

func dividesBy(n int, d int) bool {
  return n % d == 0
}

File: erlang/examples/fizzbuzz.erl
-module(fizzbuzz).
-export([fizzbuzz/0]).

fizzbuzz() ->
  lists:map(fun(X) -> fb(X) end, lists:seq(1, 100)).

fb(X) ->
  if
    X rem 15 =:= 0 -> io:fwrite("FizzBuzz~n");
    X rem 3 =:= 0 -> io:fwrite("Fizz~n");
    X rem 5 =:= 0 -> io:fwrite("Buzz~n");
    true        -> io:fwrite("~w~n",[X])
  end.